Unable to build on Ubuntu 18.04

I'm struggling to build on Ubuntu 18.04 recently. Things get going for a few minutes and then:

<details>
<summary>MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.</summary>
<pre>
System.ArgumentNullException: Value cannot be null.
Parameter name: array
   at Microsoft.Build.Execution.BuildManager.EndBuild()
   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Int32 cpuCount, Boolean enableNodeReuse, TextWriter preprocessWriter, Boolean detailedSummary, ISet`1 warningsAsErrors, ISet`1 warningsAsMessages, Boolean enableRestore, ProfilerLogger profilerLogger, Boolean enableProfiler, Boolean interactive, Boolean isolateProjects, Boolean graphBuild, String[] inputResultsCaches, String outputResultsCache)
</pre>
</details>

I'm not sure if this is related to #37868 or not, but on the surface it seems like two different issues.

This error always occurs after this in the build log:

```
Verifying closure of runtime.linux-x64.Microsoft.Private.CoreFx.NETCoreApp runtime assemblies
Verifying no duplicate types in runtime.linux-x64.Microsoft.Private.CoreFx.NETCoreApp runtime assemblies
Microsoft.Private.CoreFx.NETCoreApp -> /code/personal/corefx/artifacts/packages/Debug/specs/runtime.linux-x64.Microsoft.Private.CoreFx.NETCoreApp.nuspec
runtime.native.System.IO.Ports -> /code/personal/corefx/artifacts/packages/Debug/specs/runtime.native.System.IO.Ports.nuspec
runtime.native.System.IO.Ports -> /code/personal/corefx/artifacts/packages/Debug/specs/runtime.linux-x64.runtime.native.System.IO.Ports.nuspec
💥
```

For the simple project, without vscode running I get:
```
[tmds@rex testproj]$ ~/repos/corefx/.dotnet/dotnet build test.proj
Microsoft (R) Build Engine version 16.1.67-preview+g13843078ee for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Inner for a
  Inner for b

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.36
```
log:
```
 (TID 10) 636951877759669195 +    0.5351ms: Starting to acquire a new or existing node to establish node ID 2...
 (TID 10) 636951877759805268 +   13.6073ms: Attempting to connect to each existing msbuild.exe process in turn to establish node 2...
 (TID 10) 636951877759826575 +    2.1307ms: Attempting connect to PID 9843 with pipe MSBuild9843 with timeout 0 ms
 (TID 10) 636951877759937305 +    11.073ms: Writing handshake to pipe MSBuild9843
 (TID 10) 636951877759965521 +    2.8216ms: Reading handshake from pipe MSBuild9843
 (TID 10) 636951877759989068 +    2.3547ms: Successfully connected to pipe MSBuild9843...!
 (TID 10) 636951877759989569 +    0.0501ms: Successfully connected to existed node 2 which is PID 9843
```
With vscode running:
```
[tmds@rex testproj]$ ~/repos/corefx/.dotnet/dotnet build test.proj
Microsoft (R) Build Engine version 16.1.67-preview+g13843078ee for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Inner for a
  Inner for b

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.36
MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
System.ArgumentNullException: Value cannot be null.
Parameter name: array
   at Microsoft.Build.Execution.BuildManager.EndBuild()
   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Int32 cpuCount, Boolean enableNodeReuse, TextWriter preprocessWriter, Boolean detailedSummary, ISet`1 warningsAsErrors, ISet`1 warningsAsMessages, Boolean enableRestore, ProfilerLogger profilerLogger, Boolean enableProfiler, Boolean interactive, Boolean isolateProjects, Boolean graphBuild, String[] inputResultsCaches, String outputResultsCache)

MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
System.ArgumentNullException: Value cannot be null.
Parameter name: array
   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Int32 cpuCount, Boolean enableNodeReuse, TextWriter preprocessWriter, Boolean detailedSummary, ISet`1 warningsAsErrors, ISet`1 warningsAsMessages, Boolean enableRestore, ProfilerLogger profilerLogger, Boolean enableProfiler, Boolean interactive, Boolean isolateProjects, Boolean graphBuild, String[] inputResultsCaches, String outputResultsCache)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String[] commandLine)

Unhandled Exception: System.ArgumentNullException: Value cannot be null.
Parameter name: array
   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Int32 cpuCount, Boolean enableNodeReuse, TextWriter preprocessWriter, Boolean detailedSummary, ISet`1 warningsAsErrors, ISet`1 warningsAsMessages, Boolean enableRestore, ProfilerLogger profilerLogger, Boolean enableProfiler, Boolean interactive, Boolean isolateProjects, Boolean graphBuild, String[] inputResultsCaches, String outputResultsCache)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String[] commandLine)
   at Microsoft.Build.CommandLine.MSBuildApp.Main(String[] args)
```
log:
```
 (TID 9) 636951877959420639 +    0.6439ms: Starting to acquire a new or existing node to establish node ID 2...
```

Thanks, folks! With a private of https://github.com/microsoft/msbuild/pull/4417, I get a better stack:

```
MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
System.ArgumentNullException: Value cannot be null.
Parameter name: array
   at System.Buffers.TlsOverPerCoreLockedStacksArrayPool`1.Return(T[] array, Boolean clearArray)
   at System.Diagnostics.Process.GetProcessName(Int32 processId)
   at System.Diagnostics.Process.GetProcessesByName(String processName, String machineName)
   at System.Diagnostics.Process.GetProcessesByName(String processName)
   at Microsoft.Build.BackEnd.NodeProviderOutOfProcBase.GetPossibleRunningNodes(String msbuildLocation) in /home/raines/src/msbuild/src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs:line 263
   at Microsoft.Build.BackEnd.NodeProviderOutOfProcBase.GetNode(String msbuildLocation, String commandLineArgs, Int32 nodeId, INodePacketFactory factory, Int64 hostHandshake, Int64 clientHandshake, NodeContextTerminateDelegate terminateNode) in /home/raines/src/msbuild/src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs:line 168
   at Microsoft.Build.BackEnd.NodeProviderOutOfProc.CreateNode(Int32 nodeId, INodePacketFactory factory, NodeConfiguration configuration) in /home/raines/src/msbuild/src/Build/BackEnd/Components/Communications/NodeProviderOutOfProc.cs:line 124
   at Microsoft.Build.BackEnd.NodeManager.AttemptCreateNode(INodeProvider nodeProvider, NodeConfiguration nodeConfiguration) in /home/raines/src/msbuild/src/Build/BackEnd/Components/Communications/NodeManager.cs:line 357
   at Microsoft.Build.BackEnd.NodeManager.CreateNode(NodeConfiguration configuration, NodeAffinity nodeAffinity) in /home/raines/src/msbuild/src/Build/BackEnd/Components/Communications/NodeManager.cs:line 113
   at Microsoft.Build.Execution.BuildManager.PerformSchedulingActions(IEnumerable`1 responses) in /home/raines/src/msbuild/src/Build/BackEnd/BuildManager/BuildManager.cs:line 1925
   at Microsoft.Build.Execution.BuildManager.ProcessPacket(Int32 node, INodePacket packet) in /home/raines/src/msbuild/src/Build/BackEnd/BuildManager/BuildManager.cs:line 1221
   at Microsoft.Build.Execution.BuildManager.ProcessWorkQueue(Action action) in /home/raines/src/msbuild/src/Build/BackEnd/BuildManager/BuildManager.cs:line 1174
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Build.Execution.BuildManager.EndBuild() in /home/raines/src/msbuild/src/Build/BackEnd/BuildManager/BuildManager.cs:line 778
   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Int32 cpuCount, Boolean enableNodeReuse, TextWriter preprocessWriter, Boolean detailedSummary, ISet`1 warningsAsErrors, ISet`1 warningsAsMessages, Boolean enableRestore, ProfilerLogger profilerLogger, Boolean enableProfiler, Boolean interactive, Boolean isolateProjects, Boolean graphBuild, String[] inputResultsCaches, String outputResultsCache) in /home/raines/src/msbuild/src/MSBuild/XMake.cs:line 1211
```

That looks like a corefx issue to me at first glance.

--- a/global.json
+++ b/global.json
@@ -1,6 +1,6 @@
 {
   "tools": {
-    "dotnet": "3.0.100-preview6-011681"
+    "dotnet": "3.0.100-preview6-012234"
   },
   "msbuild-sdks": {
     "Microsoft.DotNet.Arcade.Sdk": "1.0.0-beta.19302.2",

